python/ray/autoscaler/_private/autoscaler.py

```diff
@@ -452,6 +470,38 @@ def set_prometheus_updater_data(self):
                 num_recovering += 1
         self.prom_metrics.recovering_nodes.set(num_recovering)
 
+    def _report_pending_infeasible(self, resources: List[ResourceDict]):
+        """Emit event messages for infeasible or unschedulable tasks."""
```

Comment:
Can you please add more details and explanation on what is going on here and args?

Reply:
Also, shouldn't this function go into resource_demand_scheduler (why don't we change it to return the lists you need below)?

Reply:
It's better separation of concerns if RDS doesn't generate warning messages, just reports fulfilled vs unfulfilled resources.